### PR TITLE
[SP-6340] Backport of PPP-4867 - org.pentaho.reporting.engine.classic.core.allowScriptEvaluation error shown when report DOES NOT use scriptable data sources (9.3 Suite)

### PR DESCRIPTION
**Pull request by MarcDiamantHitachi**
Original PR: #1556

@bcostahitachivantara @renato-s @andreramos89

### CODE REVIEW

**Comment by smmribeiro** [comment]
@MarcDiamantHitachi there are some problems with this PR:

- The commit message should state the SP issue. Normally, we use as "[SP-6340] Backport of PPP-4867 - org.pentaho.reporting.engine.classic.core.allowScriptEvaluation error shown when report DOES NOT use scriptable data sources (9.3 Suite)"
- The final code to merge is not completely consistent to the actual code: one can see that there's extra lines on the copyright header immediately and GitHub calculates this PR as "+16 -5" while the original one (#1556) is "+7 -8"... 
- There are two commits on this PR: the "cherry pick" and a merge operation; it seems that the merge operation is "breaking" something?

Could you, please, fix these issues?

File: engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/DataFactoryScriptingSupport.java [hunk 1]
```diff
@@ -12,7 +12,9 @@
  * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  * See the GNU Lesser General Public License for more details.
  *
- * Copyright (c) 2002-2022 Hitachi Vantara..  All rights reserved.
+
+ * Copyright (c) 2002-2023 Hitachi Vantara..  All rights reserved.
+
```
Comment (smmribeiro) [inline engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/DataFactoryScriptingSupport.java:17]: Extra lines included...

File: engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/DataFactoryScriptingSupport.java [hunk 2]
```diff
@@ -448,6 +450,19 @@ public void initialize( final DataFactory dataFactory, final DataFactoryContext
...
+    boolean allowScriptEval = ClassicEngineBoot.getInstance().getGlobalConfig().getConfigProperty(
+                    "org.pentaho.reporting.engine.classic.core.allowScriptEvaluation", "false" )
+            .equalsIgnoreCase( "true" );
+
+    if ( !allowScriptEval ) {
+      DataFactoryScriptingSupport.logger.error( "Scripts are prevented from running by default in order to avoid"
+              + " potential remote code execution.  The system administrator must enable this capability by changing"
+              + " the value of org.pentaho.reporting.engine.classic.core.allowScriptEvaluation to true." );
+      return;
+    }
```
Comment (smmribeiro) [inline engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/DataFactoryScriptingSupport.java:465]: I think the merge operation is breaking something here and GitHub can't correctly calculate what to merge.

**Review by smmribeiro — CHANGES_REQUESTED** [review]

**Comment by smmribeiro** [comment]
@MarcDiamantHitachi there are some problems with this PR:

The commit message should state the SP issue. Normally, we'd use "**[SP-6340] Backport of PPP-4867 - org.pentaho.reporting.engine.classic.core.allowScriptEvaluation error shown when report DOES NOT use scriptable data sources (9.3 Suite)**"

The final code to merge is not completely consistent to the original code: one can see that there's extra lines on the copyright header immediately and GitHub calculates this PR as "+16 -5" while the original one (https://github.com/pentaho/pentaho-reporting/pull/1556) is "+7 -8"...

There are two commits on this PR: the "cherry pick" and a merge operation; I suspect that the merge operation is "breaking" something.

Could you, please, fix these issues?

**Review by andreramos89 — APPROVED** [review]

**Review by smmribeiro — APPROVED** [review]

**Pull request merged**